### ISSUE TITLE
dropout may generate different result compare to cuda

### ISSUE THREAD

**Issue by jianyizh** [intel/torch-xpu-ops#2926]
### 🐛 Describe the bug

xpu
```python
import torch
import torch.nn as nn

torch.xpu.manual_seed_all(1)

x = torch.ones(512*4*1024).to("xpu")

dropout = nn.Dropout(p=0.1,inplace=False)

dropout.train()
y_train = dropout(x)
print((y_train == 0).nonzero())
```
gets 
tensor([[      8],
        [     14],
        [     20],
        ...,
        [2097133],
        [2097145],
        [2097147]], device='xpu:0')

cuda
```python
import torch
import torch.nn as nn

torch.cuda.manual_seed_all(1)

x = torch.ones(512*4*1024).to("cuda")

dropout = nn.Dropout(p=0.1,inplace=False)

dropout.train()
y_train = dropout(x)
print((y_train == 0).nonzero())
```
gets
tensor([[      8],
        [     14],
        [     20],
        ...,
        [2097121],
        [2097140],
        [2097151]], device='cuda:0')

### Versions

bmg and 4080s
